js-core/homeworks/homework-17/main.js

```diff
@@ -0,0 +1,296 @@
+  infinity: true,
+});*/
+
+
+/*class Carousel {
+
+	static initialize() {
+		return new Carousel;
+	}
+
+	constructor(){
+		this.clickHandler();
+	}
+
+	clickHandler() {
+		let leftArrow = document.querySelector('.ec-left');
+		let rightArrow = document.querySelector('.ec-right');
+		let slides = [...document.querySelectorAll('li')];
+		let activeNumberSpan = document.querySelector('.number');
+		let activeSlide = null;
+
+		slides.forEach((item) => {
+			if (item.getAttribute('style') == 'display: block') {
+				activeSlide = item;
+				activeNumberSpan.innerHTML = activeSlide.firstElementChild.alt;
+			}
+		});
+
+		leftArrow.onclick = () => {
+			activeSlide.setAttribute('style', 'display:none');
+			if (activeSlide.previousElementSibling) {
+				activeSlide.previousElementSibling.setAttribute('style', 'display:block');
+				activeSlide = activeSlide.previousElementSibling;
+				activeNumberSpan.innerHTML = activeSlide.firstElementChild.alt;
+
+			} else {
+				slides[slides.length - 1].setAttribute('style', 'display:block');
+				activeSlide = slides[slides.length - 1];
+				activeNumberSpan.innerHTML = activeSlide.firstElementChild.alt;
+			}
+			
+		}
+
+		rightArrow.onclick = () => {
+			activeSlide.setAttribute('style', 'display:none');
+			if (activeSlide.nextElementSibling) {
+				activeSlide.nextElementSibling.setAttribute('style', 'display:block');
+				activeSlide = activeSlide.nextElementSibling;
+				activeNumberSpan.innerHTML = activeSlide.firstElementChild.alt;
+			} else {
+				slides[0].setAttribute('style', 'display:block');
+				activeSlide = slides[0];
+				activeNumberSpan.innerHTML = activeSlide.firstElementChild.alt;
+			}
+
+		}
+
+	}
+
+}
+//let myCarousel = new Carousel;
+var myInitializedCarousel = Carousel.initialize();
+*/
+
+
+ /* @SUPER -> @front-end---------------------------------------------------
+ * Уберите в стилях li - position:absolute.
+ * изменяйте свойство transform:translate3d у .carousel, добавьте transition
+ * и сделайте чтобы картинки передвигались влево и вправо
+ *
+ * @PUPER -> переход к первой картинка-------------------------------------
+ *
+ * */
+class Carousel {
+
+	static initialize() {
+		return new Carousel;
+	}
+
+	constructor(){
+		this.clickHandler();
+	}
+
+	clickHandler() {
+		let leftArrow = document.querySelector('.ec-left');
+		let rightArrow = document.querySelector('.ec-right');
+
+		let parent = document.querySelector('ul');
+		let slides = [...parent.children];
+
+		let oneChildWidth = parseInt(getComputedStyle(slides[0]).width);
+		let parentWidth = oneChildWidth * slides.length;
+		parent.style.width = parentWidth + 'px';
+
+		let activeSlide = parent.firstElementChild;
+
+		let activeNumberSpan = document.querySelector('.number');
+		activeNumberSpan.innerHTML = activeSlide.firstElementChild.alt;
+
+
+		leftArrow.onclick = () => {
+
+			if (activeSlide.nextElementSibling) {
+
+				parent.style.transform += `translateX(-${oneChildWidth}px)`;
```

Comment:
the code at line 152 looks like the same, can we make reusable function and apply it to every click?